src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5609,78 +5602,97 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
             var firstResult = new MethodGroupResolution();
             AnalyzedArguments actualArguments = null;
 
-            foreach (var scope in new ExtensionMethodScopes(this))
+            var expressionHasOmittedTypeArgs = false;
+            if (expression.Kind() == SyntaxKind.SimpleMemberAccessExpression)
             {
-                var methodGroup = MethodGroup.GetInstance();
-                var diagnostics = DiagnosticBag.GetInstance();
-
-                this.PopulateExtensionMethodsFromSingleBinder(scope, methodGroup, expression, left, methodName, typeArguments, diagnostics);
+                var memberAccessSyntax = (MemberAccessExpressionSyntax)expression;
+                if (memberAccessSyntax.Name.Kind() == SyntaxKind.GenericName)
+                {
+                    var genericNameSyntax = (GenericNameSyntax)memberAccessSyntax.Name;
+                    if (genericNameSyntax.TypeArgumentList.Arguments.Any(SyntaxKind.OmittedTypeArgument))
+                    {
+                        expressionHasOmittedTypeArgs = true;
```

Comment:
>expressionHasOmittedTypeArgs = true; [](start = 24, length = 36)

Consider simply returning from here rather than increasing nesting of the code below.